### PR TITLE
fix(DEC): MCOL-5602 fixing potentially endless loop in DEC

### PR DESCRIPTION
**Pull request by drrtuy**

### CODE REVIEW

File: dbcon/joblist/distributedenginecomm.cpp [hunk 1]
```diff
@@ -1206,9 +1181,15 @@ DistributedEngineComm::MQE::MQE(const uint32_t pCount, const uint32_t initialInt
                                 const uint64_t flowControlEnableBytesThresh)
  : ackSocketIndex(0), pmCount(pCount), hasBigMsgs(false), targetQueueSize(flowControlEnableBytesThresh)
 {
-  unackedWork.reset(new volatile uint32_t[pmCount]);
   interleaver.reset(new uint32_t[pmCount]);
-  memset((void*)unackedWork.get(), 0, pmCount * sizeof(uint32_t));
+  const uint32_t localPmCount = pmCount;
```
Comment (denis0x0D) [inline dbcon/joblist/distributedenginecomm.cpp:1185]: Didn't get why `localPmCount` is needed, if `pmCount` already exists

File: dbcon/joblist/distributedenginecomm.cpp [hunk 2]
```diff
@@ -761,65 +766,66 @@ void DistributedEngineComm::sendAcks(uint32_t uniqueID, const vector<SBS>& msgs,
   }
 }
 
-void DistributedEngineComm::nextPMToACK(boost::shared_ptr<MQE> mqe, uint32_t maxAck, uint32_t* sockIndex,
-                                        uint16_t* numToAck)
+size_t DistributedEngineComm::ackAndRotatePM(boost::shared_ptr<MQE> mqe, const size_t maxAck,
```
Comment (denis0x0D) [inline dbcon/joblist/distributedenginecomm.cpp]: Can you please explain why did you change the name of this function, it seems to me the ACK is not sending to PM from this function.

File: dbcon/joblist/distributedenginecomm.cpp (hunk 1 above)
Comment (drrtuy) [reply dbcon/joblist/distributedenginecomm.cpp:1185]: pmCount is atomic and it would be wasteful to use it in for comparison expression.

**Review by mariadb-DenisKhalikov — APPROVED** [review]
